### PR TITLE
usbc: Use a DT alias to identify USB-CPD port

### PR DESCRIPTION
**Pull request by andysan**
Samples sometimes need to identify a default USB-C port. Instead of hard-coding a node label, use a device tree alias  to point to the correct node.

The intention is that this will make it easier to include board-specific USB-C configuration (tcpc, vbus) in the main board configuration.

### CODE REVIEW

**Review by tejlmand — DISMISSED** [review]

**Review by sambhurst — DISMISSED** [review]

**Review by mbolivar-ampere — CHANGES_REQUESTED** [review]
We use aliases for conventions that multiple samples want to align on. Chosen nodes are basically for when something in subsys/ needs a hint about which peripheral among multiple options it should use.

**Comment by andysan** [comment]
> We use aliases for conventions that multiple samples want to align on. Chosen nodes are basically for when something in subsys/ needs a hint about which peripheral among multiple options it should use.

Done.

**Review by nordicjm — DISMISSED** [review]

File: samples/subsys/usb_c/sink/src/main.c [hunk 1]
```diff
@@ -13,8 +13,8 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 
-#define PORT1_NODE DT_NODELABEL(port1)
-#define PORT1_POWER_ROLE		DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
+#define PORT1_NODE		DT_ALIAS(usbc_port0)
```
Comment (keith-zephyr) [inline samples/subsys/usb_c/sink/src/main.c:16]: Update the macro naming to match the alias nodename.  You could also drop the "0" suffix, but I'm fine with or without the suffix.
```suggestion
#define USBC_PORT0_NODE		DT_ALIAS(usbc_port0)
```

File: samples/subsys/usb_c/sink/src/main.c [hunk 2]
```diff
@@ -13,8 +13,8 @@
...
+#define PORT1_POWER_ROLE	DT_ENUM_IDX(PORT1_NODE, power_role)
```
Comment (keith-zephyr) [inline samples/subsys/usb_c/sink/src/main.c:17]:
```suggestion
#define USC_PORT0_POWER_ROLE	DT_ENUM_IDX(PORT1_NODE, power_role)
```

File: samples/subsys/usb_c/source/src/main.c [hunk 3]
```diff
@@ -15,8 +15,8 @@ LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 
 #include "power_ctrl.h"
 
-#define PORT1_NODE		DT_NODELABEL(port1)
-#define PORT1_POWER_ROLE	DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
+#define PORT1_NODE		DT_ALIAS(usbc_port0)
```
Comment (keith-zephyr) [inline samples/subsys/usb_c/source/src/main.c:18]: Update this with the same naming convention that you decide on for the other sample.  `USBC_PORT0_NODE` or `USBC_PORT_NODE`.

File: samples/subsys/usb_c/sink/src/main.c (hunk 1 above)
Comment (andysan) [reply samples/subsys/usb_c/sink/src/main.c:16]: I think I prefer to keep the 0 in the alias at least. There are boards out there with multiple USB-C ports, so I think it is conceivable that we will have examples using two ports.

**Review by keith-zephyr — APPROVED** [review]

**Comment by andysan** [comment]
Ping @mbolivar-ampere

**Review by sambhurst — APPROVED** [review]

**Review by mbolivar-ampere — APPROVED** [review]

**Pull request merged**